pkg/scheduler/plugins/overcommit/overcommit.go

```diff
@@ -0,0 +1,117 @@
+	// overCommitFactor is resource overCommit factor for enqueue action
+	// It determines the number of `pending` pods that the scheduler will tolerate
+	// when the resources of the cluster is insufficient
+	overCommitFactor = "overcommit-factor"
+	// defaultOverCommitFactor defines the default overCommit resource factor for enqueue action
+	defaultOverCommitFactor = 1.2
+)
+
+type overcommitPlugin struct {
+	// Arguments given for the plugin
+	pluginArguments  framework.Arguments
+	idleResource     *api.Resource
+	inqueueResource  *api.Resource
+	overCommitFactor float64
+}
+
+// New function returns overcommit plugin object
+func New(arguments framework.Arguments) framework.Plugin {
+	return &overcommitPlugin{
+		pluginArguments:  arguments,
+		idleResource:     api.EmptyResource(),
+		overCommitFactor: defaultOverCommitFactor,
+	}
+}
+
+func (op *overcommitPlugin) Name() string {
+	return PluginName
+}
+
+func (op *overcommitPlugin) OnSessionOpen(ssn *framework.Session) {
+	overcommitStartTime := time.Now().UnixNano()
+	op.pluginArguments.GetFloat64(&op.overCommitFactor, overCommitFactor)
+	if op.overCommitFactor < 1.0 {
+		klog.Warningf("invalid input %f for overcommit-factor, reason: overcommit-factor cannot be less than 1,"+
+			" using default value: %f", op.overCommitFactor, defaultOverCommitFactor)
+		op.overCommitFactor = defaultOverCommitFactor
+	}
+	klog.V(4).Infof("overcommit plugin starts, overCommitFactor: %f", op.overCommitFactor)
+	defer klog.V(4).Infof("overcommit plugin finishes, execution time: %dns",
+		time.Now().UnixNano()-overcommitStartTime)
+
+	// calculate idle resources of total cluster, overcommit resources included
+	total := api.EmptyResource()
+	used := api.EmptyResource()
+	for _, node := range ssn.Nodes {
+		total.Add(node.Allocatable)
+		used.Add(node.Used)
+	}
+	op.idleResource = total.Clone().Multi(op.overCommitFactor).Sub(used)
+
+	// calculate inqueue job resources
+	inqueue := api.EmptyResource()
+	for _, job := range ssn.Jobs {
+		if job.PodGroup.Status.Phase == scheduling.PodGroupInqueue {
+			inqueue.Add(api.NewResource(*job.PodGroup.Spec.MinResources))
+		}
+	}
+	op.inqueueResource = inqueue.Clone()
+
+	ssn.AddJobEnqueueableFn(op.Name(), func(obj interface{}) bool {
+		job := obj.(*api.JobInfo)
+		idle := op.idleResource
+		inqueue := api.EmptyResource()
+		inqueue.Add(op.inqueueResource)
+		if job.PodGroup.Spec.MinResources == nil {
+			klog.V(4).Infof("job <%s/%s> is bestEffort, allow it be inqueue", job.Namespace, job.Name)
+			return true
+		}
+
+		//TODO: if allow 1 more job to be inqueue beyond overcommit-factor, large job may be inqueue and create pods
+		jobMinReq := api.NewResource(*job.PodGroup.Spec.MinResources)
+		if inqueue.Add(jobMinReq).LessEqual(idle) {
+			klog.V(4).Infof("sufficient resources, allow job <%s/%s> be inqueue", job.Namespace, job.Name)
+			op.inqueueResource.Add(jobMinReq)
+			return true
+		}
+		klog.V(4).Infof("idle resource in cluster is overused, ignore job <%s/%s>",
```

Comment:
change log from "idle resource in cluster is overused" to "resource in cluster is overused, not allow job <%s/%s> be inqueue"

Reply:
modified this log.